src/agentscope/file_manager.py

```diff
@@ -46,8 +46,7 @@ def _get_and_create_subdir(self, subdir: str) -> str:
         """Get the path of the subdir and create the subdir if it does not
         exist."""
         path = os.path.join(self.dir, _runtime.runtime_id, subdir)
-        if not os.path.exists(path):
```

Comment:
since we have a "if" statement here to determine the existence of the path, why do we have "File exists error" here?

Reply:
multiple threads reach this line and find the file does not exist, so they try to create the file at the same time.

Reply:
Yes